### PR TITLE
New Forge FE Instructions

### PR DESCRIPTION
**Pull request by einnesTT**
### Ticket
https://github.com/tenstorrent/tt-forge/issues/40?issue=tenstorrent%7Ctt-forge%7C148

### Problem description
Redoing TT-Forge-FE instructions so they are stable, reflect new work to create a docker image, and reflect requests for a different documentation structure

### What's changed
Test the new versions of instructions, restructured documentation, corrections for style of content to support how Marketing labels things. 

### Checklist
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: docs/src/getting_started.md [hunk 1]
```diff
@@ -1,108 +1,64 @@
...
+TT-Forge-FE can be used to run models from any framework. Because TT-Forge-FE is open source, you can also develop and add features to it. Setup instructions differ based on the task. You have the following options, listed in order of difficulty:
+* [Installing a Wheel and Running an Example](#installing-a-wheel-and-running-an-example) - You should choose this option if you want to run models.
+* [Using a Docker Container to Run an Example](getting_started_docker.md) - Choose this option if you want to keep the environment for running models separate from your existing environment.
+* [Building from Source](getting_started_build_from_source.md) - This option is best if you want to develop TT-Forge-FE further. It's a more complex process you are unlikely to need if you want to stick with running a model.
 
 ## Configuring Hardware
+Before setup can happen, you must configure your hardware. This section of the walkthrough shows you how to do a quick setup using TT-Installer.
 
-Configure your hardware with tt-installer:
+1. Configure your hardware with TT-Installer:
```
Comment (staylorTT) [inline docs/src/getting_started.md:26]: Is it worth calling out that this step only needs to be done if you have not configured your system already?

File: docs/src/getting_started.md [hunk 2]
```diff
@@ -1,108 +1,64 @@
...
+3. For this walkthrough, TT-Forge-FE is used. You need to install two wheels for set up, **forge** and **tvm**. Scroll down the releases page until you see the latest release for **tt-forge-fe nightly**.
 
-> **NOTE**: The link for each wheel you install is structured as
-`https://github.com/tenstorrent/tt-forge/releases/download/nightly-0.1.0.dev{number from top of release page}/forge-0.1.0.dev{number from top of release page}-cp310-cp310-linux_x86_64.whl`
+4. Go to the bottom of the section for your release and click on the triangle next to **Assets**. You are presented with two installable wheels, one for **tt_forge_fe** and one for **tt_tvm**. You can also download the source code for everything as a **.zip** or a **tar.gz** file.
 
-For this walkthrough, tt-forge-fe is used. You need to install two wheels for set up:
+The links listed in this walkthrough are taken from the releases page as an example, however for the most current results get the links from the [Tenstorrent Nightly Releases](https://github.com/tenstorrent/tt-forge/releases) page. Here are the example links for install:
```
[the remark this answers is not in the document]
Comment (jmcgrathTT) [reply docs/src/getting_started.md:61]: The unified is still in PR
https://github.com/tenstorrent/tt-forge/pull/166

But to your question, if you don't add a version tag it will pull the latest stable or latest nightly with `--pre` on the pip install command

File: docs/src/getting_started.md [hunk 3]
```diff
@@ -1,108 +1,64 @@
...
 
 ```bash
 pip install https://github.com/tenstorrent/tt-forge/releases/download/nightly-0.1.0.dev20250514060212/forge-0.1.0.dev20250514060212-cp310-cp310-linux_x86_64.whl
```
Comment (nvukobratTT) [inline docs/src/getting_started.md:64]: @jmcgrathTT @nsumrakTT, how can we reference the latest wheel here? 

It'll be good for the end-user to always get the latest version of the FFE wheel. That way, we can simplify the above explanation on how to get the latest, and just provide two pip commands to run. @einnesTT if there is an option to do this, we can just add a small note on how to get a specific version of our wheel.

File: docs/src/getting_started_docker.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  ghcr.io/tenstorrent/tt-forge/tt-forge-slim:dev20250618
+```
+
+5. If you want to check that it is running, open a new tab with the **Same Command** option and run the following:
+
+```bash
+docker ps
+```
+
+6. You are now ready to move on to the next section, and run your first model.
```
Comment (nvukobratTT) [inline docs/src/getting_started_docker.md:77]: Should we mention the tt-smi step here as well? Just to be certain that device is visible from inside the docker container :))

File: docs/src/getting_started_docker.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+## Running Models in Docker
```
Comment (nvukobratTT) [inline docs/src/getting_started_docker.md:79]: I would add the same instructions here as for the wheel setup. I.e., running a single op test to confirm if everything works:
```py
import torch
import forge

class Add(torch.nn.Module):
  def __init__(self):
    super().__init__()

  def forward(self, a, b):
    return a + b

a = torch.rand(size=(2, 32, 32))
b = torch.rand(size=(2, 32, 32))

framework_module = Add()
compiled_model = forge.compile(framework_module, sample_inputs=[a, b])

out = compiled_model(a, b)

print("compiled output:", out)
```

Next steps can be running a model e2e.

Comment (nvukobratTT) [reply docs/src/getting_started_docker.md:79]: Here same applies to the wheel as well. In each case, this should be the first step. Then we can have a dedicated section (or maybe a page; e.g. "Next Steps") for running a first model e2e.

File: docs/src/getting_started_docker.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+```bash
+python tt-forge/demos/tt-forge-fe/cnn/mobile_netv2_demo.py
+```
+
+4. If all goes well you will get the following output:
+
+Prediction: Samoyed, Samoyede (class 258)
+Confidence: 0.868
+
+## Where to Go Next
```
Comment (nvukobratTT) [inline docs/src/getting_started_docker.md:105]: This can be part of the new page I'm mentioning in the comment above. I.e., cloning the repo, running a model from existing tests, then running your own model.

File: docs/src/getting_started_build_from_source.md [hunk 7]
```diff
@@ -1,40 +1,81 @@
-# Building
+# Getting Started with Building from Source
```
Comment (nvukobratTT) [inline docs/src/getting_started_build_from_source.md:1]: I'll skip commenting "Building from Source" as it's still WIP. @einnesTT is planning to add more changes to this page soon :))

File: docs/src/getting_started_docker.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
+# Getting Started with Docker and Tenstorrent
```
Comment (nvukobratTT) [inline docs/src/getting_started_docker.md:1]: How are sections for getting started structured? Can we nest them under the main "Getting Started"? Similar as it's done for testing:
image

File: docs/src/running_a_custom_model.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
+# Running a Custom Model with TT-Forge-FE
```
Comment (nvukobratTT) [inline docs/src/running_a_custom_model.md:1]: What is the purpose of this page? If we're outlining next steps, we should reference the following details (based on comments from above):
1. How to find a list of available models
    - e.g. https://github.com/tenstorrent/tt-forge-fe/tree/main/forge/test/models
    - How to run them
1. How to run your model (explanation of each section)
    - It'll be good enough to explain parts from "Code Example Explanation" here. I don't think we need another example to emphasize it. Also, let's simplify the code explanation. I think it's good enough to mention that we're loading a PyTorch model in this example, and the only change the user is making is to compile the model and run it that way. The rest of the details are probably redundant for our docs as our docs.

**Comment by einnesTT** [comment]
@nvukobratTT - yes - that had a bunch of stuff for cosmetic edits, and I'm just redoing it all. Pushing that would mess things up on this branch a lot probably. (For clarity, yes this one can be closed - https://github.com/tenstorrent/tt-forge-fe/pull/2193) 

Also this isn't done yet. You can review the build file if you like, but it's a mess right now. I'm fixing it at the moment. The other files can all be looked over and commented on, including the half completed 101 page about running a model. There was talk of having things be made so more new users could participate, so that one takes your code sample and explains it in a lot of detail. If we don't like that one, we can chuck it, but I thought it might be cool to offer. People who know more can skip it.

File: docs/src/running_a_custom_model.md (hunk 9 above)
Comment (einnesTT) [reply docs/src/running_a_custom_model.md:1]: The purpose of that page (incomplete) is to break down a code sample line by line to show someone very new how it works. Terminology is explained and included. It's not for an experienced user and is designed to be skipped if you know what you're doing. It's an extra thing. There is ongoing discussion in the company about making things more accessible to more people, so this is what the page is for. The sample is repeated in the regular steps for somebody who doesn't need that info.

File: docs/src/getting_started_docker.md (hunk 8 above)
Comment (einnesTT) [reply docs/src/getting_started_docker.md:1]: I haven't gotten to that bit yet. I will need to go into the file and make sure the right file names are listed, and we can set it up in any structure. 

We can choose whatever arrangement, as long as the formatting matches most of what is currently on the Tenstorrent doc site. I said most, because unless it's been fixed (haven't checked), there are a couple places where the style is different.

File: docs/src/getting_started_build_from_source.md (hunk 7 above)
Comment (einnesTT) [reply docs/src/getting_started_build_from_source.md:1]: Yes I don't recommend that page at all at the moment!

File: docs/src/getting_started_docker.md (hunk 5 above)
Comment (einnesTT) [reply docs/src/getting_started_docker.md:79]: What would be different about doing that? Isn't it a file and you run it?

File: docs/src/getting_started.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (einnesTT) [reply docs/src/getting_started.md:61]: What does the command look like in that case? I can add it now if the PR is going to be made available live soon.

File: docs/src/getting_started_docker.md (hunk 5 above)
Comment (nsumrakTT) [reply docs/src/getting_started_docker.md:72]: Inside tt-forge there are basic_tests, demo, and benchmark (with subdirectory for each FE).
During docker build, we use basic_tests as a sanity check, then run demos to see if it actually does what is supposed to, and then benchmarks to measure performance.

File: docs/src/getting_started_docker.md (hunk 8 above)
Comment (einnesTT) [reply docs/src/getting_started_docker.md]: Got it.

File: docs/src/getting_started_build_from_source.md (hunk 7 above)
Comment (einnesTT) [reply docs/src/getting_started_build_from_source.md:1]: You should be able to comment there now.

File: docs/src/getting_started_docker.md (hunk 6 above)
Comment (einnesTT) [reply docs/src/getting_started_docker.md:110]: Check out the structure for each page.

File: docs/src/getting_started.md (hunk 3 above)
Comment (einnesTT) [reply docs/src/getting_started.md]: Fixed for now.

File: docs/src/getting_started_docker.md (hunk 5 above)
Comment (einnesTT) [reply docs/src/getting_started_docker.md:72]: I'm not sure what to do for the basic_tests etc. At the moment I have the code sample @nvukobratTT mentioned and then a section where you run a demo model from TT-Forge.

**Review by jmcgrathTT — APPROVED** [review]
Approving to fix the broken get started page

File: docs/src/getting_started_build_from_source.md [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+
+3. Please ensure that after you run this script, after you complete reboot, you activate the virtual environment it sets up - ```source ~/.tenstorrent-venv/bin/activate```.
+
+4. After your environment is running, to check that everything is configured, type the following:
+
+```bash
+tt-smi
+```
+
+You should see the Tenstorrent System Management Interface. It allows you to view real-time stats, diagnostics, and health info about your Tenstorrent device.
```
Comment (pilkicTT) [inline docs/src/getting_started_build_from_source.md:30]: this section occurs in multiple "getting started" files. can we put into into a separate file? to avoid copy-paste errors and ensure consistent instructions...

File: docs/src/getting_started.md [hunk 11]
```diff
@@ -1,168 +1,89 @@
...
-> **NOTE:** If you plan to do development work in the tt-forge repo, you need to build from source. To do that, please see the
-> [build instructions for tt-forge-fe](https://github.com/tenstorrent/tt-forge-fe/
-> blob/main/docs/src/build.md).
+## Setup Options
+TT-Forge-FE can be used to run models from any framework. Because TT-Forge-FE is open source, you can also develop and add features to it. Setup instructions differ based on the task. You have the following options, listed in order of difficulty:
+* [Installing a Wheel and Running an Example](#installing-a-wheel-and-running-an-example) - You should choose this option if you want to run models.
+* [Using a Docker Container to Run an Example](getting_started_docker.md) - Choose this option if you want to keep the environment for running models separate from your existing environment.
+* [Building from Source](getting_started_build_from_source.md) - This option is best if you want to develop TT-Forge-FE further. It's a more complex process you are unlikely to need if you want to stick with running a model.
 
 ## Configuring Hardware
```
Comment (pilkicTT) [inline docs/src/getting_started.md:23]: Since, this is reocurring, maybe we can swap the sections, so that the `Configuring Hardware` is before the `Setup Options` section.

Also, i would have instructions for wheel also follow the same convention as the other getting started files, such that it is contained in a separate file. Each file can then have a link to `Configuring Hardware` to point users if they haven't done it already.

It would look something like this:

```markdown
## Configuring Hardware
<Instructions for configuring hardware>

## Setup Options
TT-Forge-FE can be used to run models from any framework. Because TT-Forge-FE is open source, you can also develop and add features to it. Setup instructions differ based on the task. You have the following options, listed in order of difficulty:
* [Installing a Wheel and Running an Example](getting_started_wheel.md) - You should choose this option if you want to run models.
* [Using a Docker Container to Run an Example](getting_started_docker.md) - Choose this option if you want to keep the environment for running models separate from your existing environment.
* [Building from Source](getting_started_build_from_source.md) - This option is best if you want to develop TT-Forge-FE further. It's a more complex process you are unlikely to need if you want to stick with running a model.

END OF FILE
```

File: docs/src/getting_started_build_from_source.md (hunk 10 above)
Comment (pilkicTT) [reply docs/src/getting_started_build_from_source.md:30]: see the comment above, where i proposed a solution...

File: docs/src/getting_started.md [hunk 12]
```diff
@@ -1,168 +1,89 @@
...
+pip install tt_tvm  --extra-index-url https://pypi.eng.aws.tenstorrent.com/
 ```
 
-## Installing a Wheel
-This section walks you through downloading and installing a wheel. You can install the wheel wherever you would like if it's for running a model. If you want to do development work, you must clone the repo you want, navigate into it, and then set up the wheel.
-
-1. Make sure you are in an active virtual environment.
-
-> **NOTE**: If you plan to do development work, before continuing with these instructions, clone the repo you plan to use, then navigate into the repo. If you are just running models, this step is not necessary.
+3. To test that everything is running correctly, try an example model. You can use nano or another text editor to paste this code into a file named **forge_example.py** and then run it from the terminal. You should still have your virtual environment running after installing the wheels when running this example:
```
Comment (pilkicTT) [inline docs/src/getting_started.md:55]: I wouldn't explain how to write into a file:
```suggestion
3. To test that everything is running correctly, try an example model. 
3.a. Create a file `forge_example.py` with the following content:
<code snippet>
3.b. Run it with:
 ```python forge_example.py```
```

or something similar...

File: docs/src/getting_started_build_from_source.md [hunk 13]
```diff
@@ -0,0 +1,280 @@
...
+```bash
+wget -q https://github.com/dmakoviichuk-tt/mpi-ulfm/releases/download/v5.0.7-ulfm/openmpi-ulfm_5.0.7-1_amd64.deb -O /tmp/openmpi-ulfm.deb && \
+sudo apt install -y /tmp/openmpi-ulfm.deb
+```
+
+3. Export environment variables:
+
+```bash
+export PATH=/opt/openmpi-v5.0.7-ulfm/bin:$PATH
+export LD_LIBRARY_PATH=/opt/openmpi-v5.0.7-ulfm/lib:$LD_LIBRARY_PATH
```
Comment (pilkicTT) [inline docs/src/getting_started_build_from_source.md:136]: this shouldn't be necessary... it can go in for now, but i'll double check it...

also, i don't see this step in the `tt-mlir` docs...

**Review by pilkicTT — APPROVED** [review]
looks good. in the next few days, i'll test "build from source" instructions on a bare metal  machine, to double check everything.

**Review by jmcgrathTT — APPROVED** [review]
